sycl/source/detail/graph_impl.cpp

```diff
@@ -343,6 +343,9 @@ void exec_graph_impl::createCommandBuffers(sycl::device Device) {
 }
 
 exec_graph_impl::~exec_graph_impl() {
+  WriteLock Lock(MMutex);
+  WriteLock LockImpl(MGraphImpl->MMutex);
```

Comment:
Do we need to hold this lock for the entirety of this function or could we let go of it as soon as `MGraphImpl->clearQueues()` is done?

Reply:
You are right. No need to lock the Mutex for the entirety of this function.

Reply:
Indeed, the MGraphImpl mutex is used to protect the graph_impl object from concurrent accesses but also the nodes that make up this graph.
Given that these same nodes are referenced by other elements such as MSchedule, we need to release all references to those nodes under exclusive access to avoid memory leaks.